File: app/src/main/java/edu/gdei/gdeiassistant/Tools/TimeUtils.java
package edu.gdei.gdeiassistant.Tools;

import java.util.Date;

public class TimeUtils {

    public enum TimeUnit {

        SECOND(0), MINUTES(1), HOUR(2), DAY(3);

        TimeUnit(Integer unit) {
            this.unit = unit;
        }

        private Integer unit;

        public Integer getUnit() {
            return unit;
        }
    }

    /**
     * 两个时间间的时间戳计算函数
     *
     * @param beginDate
     * @param endDate
     * @param timeUnit
     * @return
     */
    public static long GetTimestampDifference(Date beginDate, Date endDate, TimeUnit timeUnit) {
        if (beginDate == null || endDate == null) {
            return 0;
        }
        long millisecond = endDate.getTime() - beginDate.getTime();
        switch (timeUnit.unit) {
            case 0:
                return (millisecond / 1000);
            case 1:
                return (millisecond / (1000 * 60));
            case 2:
                return (millisecond / (1000 * 60 * 60));
            case 3:
                return (millisecond / (1000 * 60 * 60 * 24));
        }
        return 0;
    }
}

File: app/src/main/java/cn/gdeiassistant/util/TimeUtils.kt
package cn.gdeiassistant.util

import java.util.Date

object TimeUtils {
    enum class TimeUnit(val unit: Int) { SECOND(0), MINUTES(1), HOUR(2), DAY(3) }
    @JvmStatic
    fun GetTimestampDifference(beginDate: Date?, endDate: Date?, timeUnit: TimeUnit): Long {
        if (beginDate == null || endDate == null) return 0L
        val millis = endDate.time - beginDate.time
        return when (timeUnit.unit) {
            0 -> millis / 1000
            1 -> millis / (1000 * 60)
            2 -> millis / (1000 * 60 * 60)
            3 -> millis / (1000 * 60 * 60 * 24)
            else -> 0L
        }
    }
}
